### PR TITLE
Created Delete Bill API

### LINKED ISSUE #140: Feature Request: Delete bills
**User Story**
For example: 
As a user
I want to add this feature
So this can happen

**Priority**
[x]: High
[]: Medium
[]: Low

**Describe the solution you want to implement**
 * Only the payee can delete/archive the bill

**Describe alternatives you've considered**

**Issue Dependencies**
Does this issue depend on any other issues?

**Additional context**

### PR DESCRIPTION
**Pull request by TheLastPiixel**
Closes #(issue)

Delete Bill API created in bill_exec.py

Write a simple description of the added feature or solution to the bug.
- you can have a bullet list with distinct bits of functionality
- a good thing to mention here would be API calls made or DB tables accessed, etc.

**Testing**

**Checklist**

- [X] I have tested my change
- [X] Code builds successfully and all tests pass
- [x] Docs have been added/updated if needed

### CODE REVIEW

File: server/src/db_manager.py [hunk 1]
```diff
@@ -325,3 +325,57 @@ def create_test_data(self):
...
+        except Exception as e:
+            print(e)
+            return False
+        finally:
+            try:
+                conn.close()
+            except UnboundLocalError:
+                pass
+
+    def check_valid_user(self, username : str, bill_id : int):
```
Comment (andrewh318) [inline server/src/db_manager.py]: I think this method name is a bit ambiguous due to 'valid user'.

We are essentially checking if the user owns the bill here right (and hence, has permission to delete)? How about `check_if_user_owns_bill`? Or something similar :)

**Review by andrewh318 — CHANGES_REQUESTED** [review]

File: server/src/db_manager.py (hunk 1 above)
Comment (kaplunket) [reply server/src/db_manager.py]: Yeah, I agree that a more descriptive name would be nice.

Comment (TheLastPiixel) [reply server/src/db_manager.py]: Cool guys, thanks for the feedback. I've since made the changes.

**Review by kaplunket — APPROVED** [review]
Looks good to me

**Review by GrunWong — APPROVED** [review]
LGTM

**Review by andrewh318 — APPROVED** [review]

**Pull request merged**